app/controllers/returns/credits/ClaimForWhichYearController.scala

```diff
@@ -50,14 +50,12 @@ class ClaimForWhichYearController @Inject()(
 
   def onPageLoad(mode: Mode): Action[AnyContent] =
     journeyAction.async { implicit request =>
-      availableRemainingOptions.map {
-        options =>
-        if (options.isEmpty) {
-          Redirect(controllers.returns.credits.routes.CreditsClaimedListController.onPageLoad(mode))
-        } else {
+      availableRemainingOptions.flatMap {
+        case Nil => Future.successful(Redirect(controllers.returns.credits.routes.CreditsClaimedListController.onPageLoad(mode)))
+        case Seq(onlyOption) => selectDateRange(onlyOption, mode)
```

Comment:
This has a bug! 
By updating the useranswers twice you loose the first lot of sets. as it is not passed through updates -.- Need to rethink this.